### PR TITLE
compares: support Same/NotSame

### PR DESCRIPTION
**Pull request by Antonboom**
Closes https://github.com/Antonboom/testifylint/issues/93

### CODE REVIEW

**Comment by ccoVeille** [comment]
I'll check later today, but the code looks good.

File: internal/checkers/compares.go [hunk 1]
```diff
@@ -42,13 +46,24 @@ func (checker Compares) Check(pass *analysis.Pass, call *CallMeta) *analysis.Dia
...
-		tokenToProposedFn = tokenToProposedFnInsteadOfTrue
+		if ptrComparison {
+			tokenToProposedFn = tokenToProposedFnInsteadOfTrueForPtr
+		} else {
+			tokenToProposedFn = tokenToProposedFnInsteadOfTrue
+		}
+
 	case "False":
-		tokenToProposedFn = tokenToProposedFnInsteadOfFalse
+		if ptrComparison {
```
Comment (ccoVeille) [inline internal/checkers/compares.go:62]: This is the kind of implementation I was thinking for promoting zero over empty

https://github.com/Antonboom/testifylint/issues/75#issuecomment-2041380821

https://github.com/Antonboom/testifylint/issues/75#issuecomment-2067622523

File: internal/checkers/compares.go [hunk 2]
```diff
@@ -42,13 +46,24 @@ func (checker Compares) Check(pass *analysis.Pass, call *CallMeta) *analysis.Dia
 		return nil
 	}
 
+	_, ptrComparison := pass.TypesInfo.TypeOf(be.X).(*types.Pointer)
+
 	var tokenToProposedFn map[token.Token]string
 
 	switch call.Fn.NameFTrimmed {
```
Comment (ccoVeille) [inline internal/checkers/compares.go:53]: This code works, but then it becomes very difficult to read. I mean a switch with 2 if in it.

Maybe code could be refactored to use a function to return the token suggestion.

Or the map, could have extra level.

Just a suggestion, you can of course discard

**Review by ccoVeille — APPROVED** [review]

**Comment by ccoVeille** [comment]
I confirm it's OK, except tge few remarks I raised but that could be ignored, or refactored later

File: internal/checkers/compares.go (hunk 1 above)
Comment (Antonboom) [reply internal/checkers/compares.go:62]: I have a "zero thread" in to-do :sorry:
complex story 😢

File: internal/checkers/compares.go (hunk 2 above)
Comment (Antonboom) [reply internal/checkers/compares.go:53]: > Or the map, could have extra level.

I thought about it, but did not find the appropriate extendable key

But I simplified the code, thanks 👌

File: internal/checkers/compares.go (hunk 1 above)
Comment (ccoVeille) [reply internal/checkers/compares.go]: OK, understood. I'm not very available either, haha

File: internal/checkers/compares.go (hunk 2 above)
Comment (ccoVeille) [reply internal/checkers/compares.go]: I confirm, it's way better now. The code is good and clear. I don't think the map thing would have been as simple as your implementation. So I'm very happy with it.

**Pull request merged**